### PR TITLE
[Docs] Add docs/reference split

### PR DESCRIPTION
**Pull request by ErikSchierboom**

### CODE REVIEW

File: languages/csharp/docs/README.md [hunk 1]
```diff
@@ -1,8 +1,20 @@
 # C# specific docs
 
-## General
-- [How to implement a Concept Exercise](./implementing-a-concept-exercise.md)
+## Reference docs
```
Comment (iHiD) [inline languages/csharp/docs/README.md]: I suggest calling this something different to "reference docs" for clarity. Maybe "Student-facing docs"?

Comment (ErikSchierboom) [reply languages/csharp/docs/README.md]: Excellent!

Comment (ErikSchierboom) [reply languages/csharp/docs/README.md]: Updated. We should probably rename the "new reference document" GitHub issue template. Agreed? @iHiD

File: .github/ISSUE_TEMPLATE/new-reference-doc.md [hunk 2]
```diff
@@ -1,42 +1,32 @@
...
-[A clear description of what information the reference document aims to teach. It should be clear how this document can help a student.
-
-For example:
-
-There are multiple uses of feature `X` in language `Y`. While...]
-
-## Resources to refer to
-
-[List additional rescources to link to from the document.
+[A clear description of what information the reference document should contain. It should be clear how this document can help a student.
```
Comment (iHiD) [inline .github/ISSUE_TEMPLATE/new-reference-doc.md]:
```suggestion
[A clear description of what information the reference document should contain. This document should help other maintainers developing and building context exercises
```

Maybe?

File: .github/ISSUE_TEMPLATE/new-reference-doc.md [hunk 3]
```diff
@@ -1,42 +1,32 @@
...
-[A clear description of what information the reference document aims to teach. It should be clear how this document can help a student.
-
-For example:
-
-There are multiple uses of feature `X` in language `Y`. While...]
-
-## Resources to refer to
-
-[List additional rescources to link to from the document.
+[A clear description of what information the reference document should contain. This document should help other maintainers developing and building context exercises
```
Comment (iHiD) [inline .github/ISSUE_TEMPLATE/new-reference-doc.md]:
```suggestion
[A clear description of what information the reference document should contain. This document should help other maintainers developing and building context exercises.
```

Sorry :)

File: .github/ISSUE_TEMPLATE/new-reference-doc.md [hunk 4]
```diff
@@ -1,42 +1,32 @@
...
 ### Contributing
 
 [Describe the exact steps the contributor has to take to add the document.
 
 For example:
 
-- Create the document using ../new/master?filename=languages/&lt;lang&gt;/reference/&lt;doc&gt;.md.
-- Add a link to the reference document in the reference README at ../tree/master/languages/&lt;lang&gt;/reference/readme.md.
+- Create the document in the appropriate location in the `reference` folder.
+- Check to see if there are other document that can refer to the new document.
```
Comment (kotp) [inline .github/ISSUE_TEMPLATE/new-reference-doc.md]:
```suggestion
- Check to see if there are other documents that can refer to the new document.
```

File: .github/ISSUE_TEMPLATE/new-student-facing-doc.md [hunk 5]
```diff
@@ -0,0 +1,40 @@
+---
+
+name: "[MAINTAINERS] Add new student-facing document"
+about: FOR MAINTAINERS ONLY - Use this template to create an issue to add a new student-facing document
+title: "[<LANG>] Add new student-facing document: <DOC>"
+labels: status/help-wanted, type/new-student-facing-doc
+assignees: ""
+---[Use this issue template to describe how to add a new student-facing doc. Only suggest adding a new student-facing doc when no appropriate online resource can be found. Please fill in the issue template, and remove any sections wrapped in square brackets (such as this section!).]
```
Comment (kotp) [inline .github/ISSUE_TEMPLATE/new-student-facing-doc.md]:
```suggestion
---[Use this issue template to describe how to add a new student-facing document. Only suggest adding a new student-facing document when no appropriate online resource can be found. Please fill in the issue template, and remove any sections wrapped in square brackets (such as this section!).]
```

File: languages/csharp/docs/README.md [hunk 6]
```diff
@@ -1,8 +1,20 @@
 # C# specific docs
 
-## General
-- [How to implement a Concept Exercise](./implementing-a-concept-exercise.md)
+## Student-facing docs
 
-## Examples
-- [New concept exercise (Arrays)](examples/new-concept-exercise-arrays.md)
-- [New reference doc (readonly vs const)](examples/new-reference-doc-readonly-vs-const.md)
+Student-facing docs are written to help explain a particular C# concept to a student when no appropriate online document can be found. They will be used when creating exercises and as references in exercise documentation.
```
Comment (kotp) [inline languages/csharp/docs/README.md]:
```suggestion
Student-facing documents are written to help explain a particular C# concept to a student when no appropriate online document can be found. They will be used when creating exercises and as references in exercise documentation.
```

File: languages/csharp/docs/README.md [hunk 7]
```diff
@@ -1,8 +1,20 @@
...
+
+The following student-facing docs have been written:
```
Comment (kotp) [inline languages/csharp/docs/README.md]:
```suggestion
The following student-facing documents have been written:
```

File: languages/csharp/docs/README.md [hunk 8]
```diff
@@ -1,8 +1,20 @@
...
+
+- [Assemblies][assemblies]
+- [Code style][code_style]
+- [Memory allocation][memory_allocation]
+
+The following student-facing docs should be written:
```
Comment (kotp) [inline languages/csharp/docs/README.md]:
```suggestion
The following student-facing documents should be written:
```

File: .github/ISSUE_TEMPLATE/new-reference-doc.md [hunk 9]
```diff
@@ -1,42 +1,32 @@
...
 
-[Use this issue template to describe how to add a new reference doc. Only suggest adding a new reference doc when no appropriate online resource can be found. Please fill in the issue template, and remove any sections wrapped in square brackets (such as this section!).]
+name: "[MAINTAINERS] Add new reference document"
+about:
+FOR MAINTAINERS ONLY - Use this template to create an issue to add a new reference
+document
+title: "[Docs] Add new reference document: <DOC>"
+labels: status/help-wanted, type/new-reference-doc
+assignees: ""
+---[Use this issue template to describe how to add a new reference doc. Only suggest adding a new reference doc when no existing reference document has been written. Please fill in the issue template, and remove any sections wrapped in square brackets (such as this section!).]
```
Comment (SaschaMann) [inline .github/ISSUE_TEMPLATE/new-reference-doc.md]:
```suggestion
---

[Use this issue template to describe how to add a new reference doc. Only suggest adding a new reference doc when no existing reference document has been written. Please fill in the issue template, and remove any sections wrapped in square brackets (such as this section!).]
```

Not sure if Markdown frontmatter will work if there's text in the same line as the closing `---`.

File: languages/fsharp/docs/README.md [hunk 10]
```diff
@@ -1,5 +1,17 @@
 # F# specific docs
 
-## General
+## Student-facing docs
 
-- [How to implement a Concept Exercise](./implementing-a-concept-exercise.md)
+Student-facing docs are written to help explain a particular F# concept to a student when no appropriate online document can be found. They will be used when creating exercises and as references in exercise documentation.
```
Comment (kotp) [inline languages/fsharp/docs/README.md]:
```suggestion
Student-facing documents are written to help explain a particular F# concept to a student when no appropriate online document can be found. They will be used when creating exercises and as references in exercise documentation.
```

File: languages/fsharp/docs/README.md [hunk 11]
```diff
@@ -1,5 +1,17 @@
...
+
+The following student-facing docs have been written:
```
Comment (kotp) [inline languages/fsharp/docs/README.md]:
```suggestion
The following student-facing documents have been written:
```

File: languages/fsharp/docs/README.md [hunk 12]
```diff
@@ -1,5 +1,17 @@
...
+
+- [Assemblies][assemblies]
+
+The following student-facing docs should be written:
```
Comment (kotp) [inline languages/fsharp/docs/README.md]:
```suggestion
The following student-facing documents should be written:
```

**Comment by ErikSchierboom** [comment]
Thanks for the reviews everyone!

**Pull request merged**